src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/SentencingSynchronisationIntTest.kt

```diff
@@ -0,0 +1,445 @@
+
+        courtSentencingNomisApiMockServer.stubGetSentence(
+          bookingId = NOMIS_BOOKING_ID,
+          sentenceSequence = NOMIS_SENTENCE_SEQUENCE,
+        )
+        dpsCourtSentencingServer.stubPostSentenceForCreate(sentenceId = DPS_SENTENCE_ID)
+
+        courtSentencingMappingApiMockServer.stubSentenceMappingCreateConflict(
+          existingDpsSentenceId = EXISTING_DPS_SENTENCE_ID,
+          duplicateDpsSentenceId = DPS_SENTENCE_ID,
+          nomisSentenceSequence = NOMIS_SENTENCE_SEQUENCE,
+        )
+
+        awsSqsCourtSentencingOffenderEventsClient.sendMessage(
+          courtSentencingQueueOffenderEventsUrl,
+          sentenceEvent(
+            eventType = "OFFENDER_SENTENCES-INSERTED",
+            sentenceSequence = NOMIS_SENTENCE_SEQUENCE,
+            bookingId = NOMIS_BOOKING_ID,
+            offenderNo = OFFENDER_ID_DISPLAY,
+          ),
+        )
+
+        // wait for mapping calls before verifying
+        await untilAsserted {
+          courtSentencingMappingApiMockServer.verify(
+            1,
+            postRequestedFor(urlPathEqualTo("/mapping/court-sentencing/sentences")),
+          )
+        }
+
+        // doesn't retry
+        dpsCourtSentencingServer.verify(
+          1,
+          postRequestedFor(urlPathEqualTo("/sentence")),
+        )
+
+        await untilAsserted {
+          verify(telemetryClient).trackEvent(
+            org.mockito.kotlin.eq("from-nomis-sync-sentence-duplicate"),
+            check {
+              assertThat(it["migrationId"]).isNull()
+              assertThat(it["existingDpsSentenceId"]).isEqualTo(EXISTING_DPS_SENTENCE_ID)
+              assertThat(it["duplicateDpsSentenceId"]).isEqualTo(DPS_SENTENCE_ID)
+              assertThat(it["existingNomisSentenceSequence"]).isEqualTo(NOMIS_SENTENCE_SEQUENCE.toString())
+              assertThat(it["duplicateNomisSentenceSequence"]).isEqualTo(NOMIS_SENTENCE_SEQUENCE.toString())
+            },
+            isNull(),
+          )
+        }
+      }
+    }
+  }
+
+  private fun waitForTelemetry(times: VerificationMode = atLeastOnce()) {
+    await untilAsserted {
+      verify(telemetryClient, times).trackEvent(
+        any(),
+        any(),
+        isNull(),
+      )
+    }
+  }
+}
+
+fun sentenceEvent(
+  eventType: String,
+  bookingId: Long = NOMIS_BOOKING_ID,
+  sentenceSequence: Int = NOMIS_SENTENCE_SEQUENCE,
+  offenderNo: String = OFFENDER_ID_DISPLAY,
+  auditModule: String = "DPS",
+) = """{
+    "MessageId": "ae06c49e-1f41-4b9f-b2f2-dcca610d02cd", "Type": "Notification", "Timestamp": "2019-10-21T14:01:18.500Z", 
+    "Message": "{\"eventId\":\"5958295\",\"eventType\":\"$eventType\",\"eventDatetime\":\"2019-10-21T15:00:25.489964\",\"bookingId\": \"$bookingId\",\"sentenceSequence\": \"$sentenceSequence\",\"offenderIdDisplay\": \"$offenderNo\",\"nomisEventType\":\"COURT_EVENT\",\"auditModuleName\":\"$auditModule\" }",
+    "TopicArn": "arn:aws:sns:eu-west-1:000000000000:offender_events", 
```

Comment:
tiny almost irrelevant comment: maybe set `nomisEventType` to eventType given they will be same for these events just in case some future person gets confused